test/cypress/integration/widgets/filters_spec.ts

```diff
@@ -243,4 +269,433 @@ describe('Filters widget', () => {
+                .createSites([
+                    { name: `${testPrefix}_Cracow` },
+                    { name: `${testPrefix}_Warsaw` },
+                    { name: `${testPrefix}_TelAviv` }
+                ]);
+        });
+
+        function selectRuleRowType(ruleRowType: FilterRuleRowType) {
+            withinLastRuleRow(() => {
+                cy.get('div[name="ruleRowType"]').click();
+                cy.get(`div[option-value="${ruleRowType}"]`).click();
+            });
+        }
+
+        function selectRuleOperator(operator: FilterRuleOperator) {
+            withinLastRuleRow(() => {
+                cy.get('div[name="ruleOperator"]').click();
+                cy.get(`div[option-value="${operator}"]`).click();
+            });
+        }
+
+        function selectRuleAttributeValues(
+            ruleRowType: FilterRuleRowType,
+            values: string[],
+            newValues = [] as string[],
+            withAutocomplete = false
+        ) {
+            if (values.length > 0) {
+                withinLastRuleRow(() => {
+                    const searchEndpoint: Record<FilterRuleRowType, string> = {
+                        blueprint_id: 'blueprints',
+                        site_name: 'sites',
+                        created_by: 'users',
+                        label: '' // NOTE: Only endpoints for attribute rules are necessary
+                    };
+                    cy.get('div[name="ruleValue"]')
+                        .click()
+                        .within(() => {
+                            values.forEach(value => {
+                                if (withAutocomplete) {
+                                    const searchEndpointRegExp = RegExp(
+                                        `${searchEndpoint[ruleRowType]}.*_search=${value}`
+                                    );
+                                    cy.interceptSp('GET', searchEndpointRegExp).as(`valueSearch_${value}`);
+                                }
+                                cy.get('input').type(`${value}`);
+                                if (withAutocomplete) cy.wait(`@valueSearch_${value}`);
+                                if (newValues.includes(value)) cy.contains('[role="option"]', 'Add ').click();
+                                else cy.get(`div[option-value="${value}"]`).click();
+                                cy.get(`.label[value="${value}"]`).should('exist');
+                            });
+                        });
+                });
+            }
+        }
+
+        function selectRuleLabelKey(key: string, newKey = false) {
+            withinLastRuleRow(() => {
+                cy.interceptSp('GET', `/labels/deployments?_search=${key}`).as(`keySearch_${key}`);
+                cy.get('div[name="labelKey"]').within(() => {
+                    cy.get('input').type(key);
+                    cy.wait(`@keySearch_${key}`);
+                    if (newKey) cy.contains('[role="option"]', 'New key ').click();
```

Comment:
Is the space at the end of `New key` significant?

```suggestion
                    if (newKey) cy.contains('[role="option"]', 'New key').click();
```

Reply:
I would like to leave it, it clearly maps to the dropdown setting. Empty space is expected, then comes the key, but as it is inside `<b>...</b>` simply having ``'New key ${newKey}`` did not work.

Reply:
Oh, so it does matter :slightly_smiling_face: I didn't figure it out just by reading the test's code, but I guess it does not matter much